gravatar/src/test/java/com/gravatar/AvatarServiceTest.kt

```diff
@@ -29,8 +32,10 @@ class AvatarServiceTest {
 
     private lateinit var avatarService: AvatarService
 
+    @OptIn(ExperimentalCoroutinesApi::class)
     @Before
     fun setUp() {
+        Dispatchers.setMain(containerRule.testDispatcher)
```

Comment:
Not sure if it makes sense to set this in `GravatarSdkContainerRule` Also it's a good practice to do a `Dispatchers.resetMain()` call after the test finishes running.

```kotlin
...
override fun evaluate() {
    try {
        gravatarSdkContainerMock = mockk<GravatarSdkContainer>()
        gravatarApiServiceMock = mockk<GravatarApiService>(relaxed = true)
        mockkObject(GravatarSdkContainer)
        every { GravatarSdkContainer.instance } returns gravatarSdkContainerMock
        every { gravatarSdkContainerMock.getGravatarApiService(any()) } returns gravatarApiServiceMock
        Dispatchers.setMain(testDispatcher)
        
        base.evaluate()
    } finally {
        Dispatchers.resetMain()
    }
}
...            
```

Reply:
Alternatively you can make a new reusable JUnit4 TestRule to set and reset the main dispatcher. See: https://developer.android.com/kotlin/coroutines/test#setting-main-dispatcher Specifically the sample class `MainDispatcherRule` in the link.